ipatests/test_integration/test_ipa_ipa_migration.py

```diff
@@ -946,3 +891,416 @@ def test_ipa_hbac_rule_duplication(self):
+            "cn=Directory Manager",
+            self.master.config.admin_password,
+            extra_args=['-n'],
+        )
+        tasks.clear_sssd_cache(self.master)
+        result1 = self.master.run_command(['id', username])
+        tasks.clear_sssd_cache(self.replicas[0])
+        result2 = self.replicas[0].run_command(['id', username])
+        assert result1.stdout_text == result2.stdout_text
+
+
+class TestIPAMigrationWithADtrust(IntegrationTest):
+    """
+    Test for ipa-migrate tool with IPA Master having trust setup
+    with Windows AD.
+    """
+    topology = "line"
+    num_ad_domains = 1
+    num_replicas = 1
+
+    @classmethod
+    def install(cls, mh):
+        tasks.install_master(
+            cls.master, setup_dns=True, extra_args=['--no-dnssec-validation']
+        )
+        cls.ad = cls.ads[0]
+        cls.ad_domain = cls.ad.domain.name
+        tasks.install_adtrust(cls.master)
+        tasks.configure_dns_for_trust(cls.master, cls.ad)
+        tasks.establish_trust_with_ad(cls.master, cls.ad.domain.name)
+
+    def test_install_local_server(self):
+        """
+        This test installs local IPA Server() i.e new IPA server with
+        the same realm and domain name that will receive the migration data.
+        """
+        tasks.install_master(self.replicas[0], setup_dns=True, setup_kra=True)
+        tasks.install_adtrust(self.replicas[0])
+
+    def test_check_ad_attributes_migrate_prod_mode(self):
+        """
+        This test checks that IPA-AD trust related attributes
+        are migrated to local server.
+        """
+        base_dn = str(self.master.domain.basedn)
+        TRUST_LOG = (
+            "Added entry: cn={},cn=ad,cn=trusts,{}"
+        ).format(self.ad_domain, base_dn)
+        CMD_LOG = "Trust direction: Trusting forest"
+        run_migrate(
+            self.replicas[0],
+            "prod-mode",
+            self.master.hostname,
+            "cn=Directory Manager",
+            self.master.config.admin_password,
+            extra_args=['-n']
+        )
+        cmd = self.replicas[0].run_command(
+            ['ipa', 'trust-show', self.ad_domain]
+        )
+        assert CMD_LOG in cmd.stdout_text
+        install_msg = self.replicas[0].get_file_contents(
+            paths.IPA_MIGRATE_LOG, encoding="utf-8"
+        )
+        assert TRUST_LOG in install_msg
+
+    def test_check_domain_sid_is_migrated(self):
+        """
+        This testcase checks that domain sid is
+        mirgrated from a remote server having trust with AD
+        to local server and is displayed in the
+        ipa trustconfig-show command
+        """
+        cmd1 = self.master.run_command(["ipa", "trustconfig-show"])
+        lines = cmd1.stdout_text.splitlines()
+        sid_remote = lines[1].lstrip()
+        cmd2 = self.replicas[0].run_command(
+            ["ipa", "trustconfig-show"]
+        )
+        lines2 = cmd2.stdout_text.splitlines()
+        sid_local = lines2[1].lstrip()
```

Comment:
It is a bit fragile to rely on line number in order to extract the SID. You can do
```
import re
regexp = (r'Security Identifier: (.*)$')
sid1 = re.findall(regexp, cmd1.stdout_text, re.MULTILINE)
sid2 = re.findall(regexp, cmd2.stdout_text, re.MULTILINE)
assert sid1 == sid2
```